cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -313,30 +307,32 @@ def plot_with_plotly(x: List[float], y: List[float], labels: List[str]):
             showarrow=False,
             font=dict(size=15, color="black", family="Sans Serif"),
         )
+    logger.info("Generating the Word Embedding Plot using Flyte Deck")
     flytekit.Deck("Word Embeddings", io.to_html(fig, full_html=True))
 
 
 # %%
 # Running the Workflow
 # ====================
 #
-# Let's kick off a workflow! This will return the Word2Vec and LDA models as workflow outputs.
+# Let's kick off a workflow! This will return the inference outputs of both gensim models:
+# similar words, WMD and LDA topics.
 @workflow
-def nlp_workflow() -> workflow_outputs:
+def nlp_workflow(target_word: str = "computer") -> workflow_outputs:
```

Comment:
added this in as parameter to workflow. 
I don;t think i can pass in  param to  dataclass constructor i.e something like this

@workflow
def  nlp_workflow(target_word: str = "computer", vector_size=200):
                         corpus = generate_processed_corpus()
                         model_wv = train_word2vec_model(training_data=corpus, hyperparams=Word2VecModelHyperparams(vector_size=vector_size)). # this throws an error as `vector_size` im assuming is promise ?



```
    result = json.JSONEncoder.default(self, o)
  File "..../opt/miniconda3/envs/flyte/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Promise is not JSON serializable

  File "..../Documents/flytekit/flytekit/core/promise.py", line 937, in create_and_link_node
    raise AssertionError(f"Failed to Bind variable {k} for function {entity.name}.") from e
AssertionError: Failed to Bind variable hyperparams for function ml_training.nlp_processing.word2vec_and_lda.train_word2vec_model.
```
hyperparams=LDAModelHyperparams()

Reply:
Yes. `vector_size` will be a promise and hence, it needs to be sent to a Flyte task.